src/main/java/de/zalando/aruha/nakadi/partitioning/OrderingKeyFieldsPartitioningStrategy.java

```diff
@@ -0,0 +1,71 @@
+package de.zalando.aruha.nakadi.partitioning;
+
+import de.zalando.aruha.nakadi.domain.EventType;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import static java.lang.Math.abs;
+
+public class OrderingKeyFieldsPartitioningStrategy implements PartitioningStrategy {
+
+    @Override
+    public String calculatePartition(final EventType eventType, final JSONObject event, final int numberOfPartitions) throws InvalidOrderingKeyFieldsException {
+        final List<String> orderingKeyFields = eventType.getOrderingKeyFields();
+        if (orderingKeyFields.isEmpty()) {
+            throw new RuntimeException("Applying " + this.getClass().getSimpleName() + " although event type " +
+                    "has no ordering key fields configured.");
+        }
+
+        try {
+
+            final JsonPath traversableJsonEvent = new JsonPath(event);
+
+            final int hashValue = orderingKeyFields.stream()
+                    // The problem is that JSONObject doesn't override hashCode(). Therefore convert it to
+                    // a string first and then use hashCode()
+                    .map(okf -> traversableJsonEvent.get(okf).toString().hashCode())
+                    .mapToInt(hc -> hc)
+                    .sum();
+
+            return String.valueOf(abs(hashValue) % numberOfPartitions);
+
+        } catch (RuntimeException e) {
+            final Throwable cause = e.getCause();
+            if (cause != null && cause instanceof InvalidOrderingKeyFieldsException) {
+                throw (InvalidOrderingKeyFieldsException) cause;
```

Comment:
No, no, it's totally fine. 
I just thought that here you throw the same exception which you caught :)
But then I noticed that you throw the cause.